### PR TITLE
Fixed: Improve WebDL and WebRip Parsing

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
NO

#### Description
backport of various Radarr webdl/webrip parsing updates and sync the logic

#### Todos
- [x] Tests
- Wiki Updates

#### Issues Fixed or Closed by this PR

*

### CODE REVIEW

File: src/NzbDrone.Core/Parser/QualityParser.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public class QualityParser
 
         private static readonly Regex SourceRegex = new Regex(@"\b(?:
                                                                 (?<bluray>BluRay|Blu-Ray|HD-?DVD|BDMux|BD(?!$))|
-                                                                (?<webdl>WEB[-_. ]DL|WEBDL|AmazonHD|iTunesHD|MaxdomeHD|NetflixU?HD|WebHD|[. ]WEB[. ](?:[xh][ .]?26[45]|DDP?5[. ]1)|[. ](?-i:WEB)$|\d+0p(?:[-. ]AMZN)?[-. ]WEB[-. ]|WEB-DLMux|\b\s\/\sWEB\s\/\s\b|(?:AMZN|NF|DP)[. ]WEB[. ])|
+                                                                (?<webdl>WEB[-_. ]DL(?:mux)?|WEBDL|AmazonHD|iTunesHD|MaxdomeHD|NetflixU?HD|WebHD|[. ]WEB[. ](?:[xh][ .]?26[45]|DDP?5[. ]1)|[. ](?-i:WEB)$|(?:\d{3,4}0p)[-. ]WEB[-. ]|[-. ]WEB[-. ]\d{3,4}0p|\b\s\/\sWEB\s\/\s\b|(?:AMZN|NF|DP)[. -]WEB[. -](?!Rip))|
```
Comment (markus101) [inline src/NzbDrone.Core/Parser/QualityParser.cs:19]: This test is now failing: `into.the.Series.s03e16.h264.720p-web-handbrake.mkv`

Comment (markus101) [reply src/NzbDrone.Core/Parser/QualityParser.cs:19]: Shouldn't `\d{3,4}0p` be `\d{2,3}0p` to match `720p` or `1080p` and not `12340p`, that seems to fix the broken test as well as `Series.Title.S16.720p.WEB.DDP.5.1.H.264.PLEX` which should also pass.

**Review by markus101 — CHANGES_REQUESTED** [review]

**Review by markus101 — CHANGES_REQUESTED** [review]
The latest changes now break this test:

```
should_parse_webdl2160p_quality("Series.Title.2016.03.14.2160p.WEB.PROPER.h264-spamTV",True)
```

File: src/NzbDrone.Core/Parser/QualityParser.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public class QualityParser
 
         private static readonly Regex SourceRegex = new Regex(@"\b(?:
                                                                 (?<bluray>BluRay|Blu-Ray|HD-?DVD|BDMux|BD(?!$))|
-                                                                (?<webdl>WEB[-_. ]DL|WEBDL|AmazonHD|iTunesHD|MaxdomeHD|NetflixU?HD|WebHD|[. ]WEB[. ](?:[xh][ .]?26[45]|DDP?5[. ]1)|[. ](?-i:WEB)$|\d+0p(?:[-. ]AMZN)?[-. ]WEB[-. ]|WEB-DLMux|\b\s\/\sWEB\s\/\s\b|(?:AMZN|NF|DP)[. ]WEB[. ])|
+                                                                (?<webdl>WEB[-_. ]DL(?:mux)?|WEBDL|AmazonHD|iTunesHD|MaxdomeHD|NetflixU?HD|WebHD|[. ]WEB[. ](?:[xh][ .]?26[45]|DDP?5[. ]1)|[. ](?-i:WEB)$|(?:720p|1080p)[-. ]WEB[-. ]|[-. ]WEB[-. ](?:720p|1080p)|\b\s\/\sWEB\s\/\s\b|(?:AMZN|NF|DP)[. -]WEB[. -](?!Rip))|
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Parser/QualityParser.cs:19]:
```suggestion
                                                                (?<webdl>WEB[-_. ]DL(?:mux)?|WEBDL|AmazonHD|iTunesHD|MaxdomeHD|NetflixU?HD|WebHD|[. ]WEB[. ](?:[xh][ .]?26[45]|DDP?5[. ]1)|[. ](?-i:WEB)$|(?:720|1080|2160)p[-. ]WEB[-. ]|[-. ]WEB[-. ](?:720|1080|2160)p|\b\s\/\sWEB\s\/\s\b|(?:AMZN|NF|DP)[. -]WEB[. -](?!Rip))|
```

**Pull request merged**